### PR TITLE
replace extract function with generator

### PR DESCRIPTION
**Pull request by stood**
#97

### CODE REVIEW

**Comment by sanpii** [comment]
⚠️ This modification create a BC break, `extract()` returns an iterator instead of an array.

**Comment by stood** [comment]
I'm sorry for the name of the new function ^^

**Comment by sanpii** [comment]
Something like `to_iter()` or `into_iter()` sounds better.

File: sources/lib/ResultIterator.php [hunk 1]
```diff
@@ -276,13 +279,33 @@ public function extract()
...
      * jsonSerialize
      *
      * @see \JsonSerializable
      */
     public function jsonSerialize()
     {
-        return $this->extract();
+        $values = [];
+
+        foreach ($this->extractGenerator() as $item) {
```
Comment (Neirda24) [inline sources/lib/ResultIterator.php]: What about directly using `iterator_to_array` here ?

**Pull request closed without merging**